raster-foundry/raster-foundry
# Use d3 instead of angular-nvd3 + nvd3 + d3, misc fixes

## Overview
* Create a graphService, similar to mapService
* Rewrite the node histograms using pure d3
* Fix a bunch of redux initialization issues caused by calling subscribe in controller constructors instead of $onInit
* Wrap $element and $document in jquery because angular is being silly and using jqlite instead
* Swap out manual service injection with autoInject in a bunch of places
* Make getMap calls in controllers a normal function instead of an inline assign in the constructor. I have no idea why I did this in the first place but it's dumb.
* Move a bunch of other logic from controller constructor methods to $onInit
* Fix some imports. We should no longer be using `/* globals _, L, $, etc*/`

### Checklist
- ~Description of PR is in an appropriate section of the [changelog](https://github.com/raster-foundry/raster-foundry/blob/develop/CHANGELOG.md) and grouped with similar changes if possible~ Changelog will be updated on the anchor PR
- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo
![image](https://user-images.githubusercontent.com/4392704/47583224-c8759800-d924-11e8-9e0b-7002012a1e77.png)


## Testing Instructions
* Verify that the lab initializes correctly (no blank nodes or styles that update when you change something)
* Verify that node histograms load correctly (masks, colors, breakpoint position)
* Verify that closing and re-opening a node histogram displays correctly
* Zoom in or out in the lab and verify that histograms are not distorted
* Verify that masked areas of the histogram are shaded grey instead of a gradient
* Verify that you can load / update multiple histograms in the lab at once
* Verify that the splitter between the lab and lab map preview is movable
* Verify that changes to the histogram are correctly persisted and show on the node preview map
* verify that the gradient selector works
* verify that custom color maps still work

Closes #4201
